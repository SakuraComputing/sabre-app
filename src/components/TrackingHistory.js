import React from 'react';
import TrackingItem from './TrackingItem';

export const TrackingHistory = ({tracks}) => (

    <div>
        <h3>Parcel Tracking History</h3>

        <table className="tracking-history">
            <tbody>
                <tr>
                    <th>Status</th>
                    <th>Track Date</th>
                    <th>Service Centre</th>
                    <th>Tour</th>
                    <th>Comment</th>
                </tr>
                    {
                        (tracks.map((track) => {
                            return <TrackingItem key={track.trackDate} {...track} />;
                        }))
                    }
            </tbody>
        </table>
    </div>
);

export default TrackingHistory;