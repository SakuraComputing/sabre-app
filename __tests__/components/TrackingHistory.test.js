import React from 'react';
import { shallow } from 'enzyme';
import TrackingHistory from '../../src/components/TrackingHistory';

const tracks = [{
    status: "Out for Delivery",
    parcelId: "JD0002256504047335",
    trackDate: "5-11-2017",
    serviceCentre: "Borehamwood",
    tour: "BE45",
    comment: "On the van"
}];

describe('when tracking history renders',() => {
    test('it should render a single record correctly',() => {
        const wrapper = shallow(<TrackingHistory tracks={tracks}/>);
        expect(wrapper).toMatchSnapshot();
    });
});



