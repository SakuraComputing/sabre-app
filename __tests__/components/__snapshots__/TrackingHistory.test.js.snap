// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when tracking history renders it should render a single record correctly 1`] = `
<div>
  <h3>
    Parcel Tracking History
  </h3>
  <table
    className="tracking-history"
  >
    <tbody>
      <tr>
        <th>
          Status
        </th>
        <th>
          Track Date
        </th>
        <th>
          Service Centre
        </th>
        <th>
          Tour
        </th>
        <th>
          Comment
        </th>
      </tr>
      <TrackingItem
        comment="On the van"
        key="5-11-2017"
        parcelId="JD0002256504047335"
        serviceCentre="Borehamwood"
        status="Out for Delivery"
        tour="BE45"
        trackDate="5-11-2017"
      />
    </tbody>
  </table>
</div>
`;
